lib/curl_setup.h

```diff
@@ -637,19 +637,22 @@ int netware_init(void);
 
 /* Single point where USE_NTLM definition might be defined */
 #if !defined(CURL_DISABLE_NTLM) && !defined(CURL_DISABLE_CRYPTO_AUTH)
-#if defined(USE_OPENSSL) || defined(USE_WINDOWS_SSPI) ||                \
+#if defined(USE_OPENSSL) || defined(USE_MBEDTLS) ||                     \
   defined(USE_GNUTLS) || defined(USE_NSS) || defined(USE_SECTRANSP) ||  \
   defined(USE_OS400CRYPTO) || defined(USE_WIN32_CRYPTO) ||              \
-  defined(USE_MBEDTLS) ||                                               \
   (defined(USE_WOLFSSL) && defined(HAVE_WOLFSSL_DES_ECB_ENCRYPT))
 
-#define USE_NTLM
+#define HAVE_NTLM_CORE
```

Comment:
1. A minor nit: we typically let configure set the defines starting with `HAVE_` and use other prefixes (like `USE_`)  for defines we set in the code that depends on things like this.

2. What is `HAVE_NTLM_CORE` vs `USE_NTLM`? How do they differ and why do we need both?

Reply:
`USE_NTLM` means that NTLM authentication in general is supported. The lower-level curl_ntlm_core functions for hash and response creation require a DES backend, which SSPI doesn't provide. They're not used by the SSPI-based NTLM authentication, but the SMB code uses them.

Reply:
Renamed to `USE_CURL_NTLM_CORE` now as that's the name of the file it guards, curl_ntlm_core.h.